src/workerd/io/worker.c++

```diff
@@ -1350,7 +1348,18 @@ Worker::Worker(kj::Own<const Script> scriptParam,
         }
       }
 
-      compileBindings(lock, *script->isolate->apiIsolate, bindingsScope);
+      kj::Maybe<v8::Local<v8::Object>> processEnv;
+      if (script->getIsolate().getApiIsolate().getFeatureFlags().getNodeJsCompat()) {
+        processEnv = v8::Object::New(lock.v8Isolate);
+      }
+
+      compileBindings(lock, *script->isolate->apiIsolate, bindingsScope, processEnv);
+
+      KJ_IF_MAYBE(env, processEnv) {
+        auto key = v8::Private::ForApi(lock.v8Isolate,
+                                       jsg::v8StrIntern(lock.v8Isolate, "processEnv"));
+        KJ_REQUIRE(jsg::check(context->Global()->SetPrivate(context, key, *env)));
```

Comment:
Note: done this way, the `process.env` object is mutable at the global scope, which matches Node.js' behavior. However, mutating `process.env` does *not* mutate the existing `env` argument passed into the handler, or used at the global scope, preserving the existing behavior. When documenting `process.env`, we'll just need to notate that `process.env` is independently mutable.